Library/Formula/crawl.rb

```diff
@@ -0,0 +1,37 @@
+class Crawl < Formula
+  desc "Dungeon Crawl: Stone Soup"
+  homepage "http://crawl.develz.org/"
+  version "0.17.1"
+
+  url "https://github.com/crawl/crawl.git",
+    :shallow => false,
+    :revision => "942a3d2973debefdaddbf6291e4ebc890725b920"
+
+  head "https://github.com/crawl/crawl.git",
+    :shallow => false
+
+  depends_on "cmake" => :build
+
+  option "with-tiles", "Build with graphical tiles"
+
+  def install
+    if build.with? "tiles"
+      extra_args = ["NO_PKGCONFIG=y", "CONTRIB_SDL=y", "TILES=y"]
+    else
+      extra_args = []
```

Comment:
Our general style here is create an `args` array with the default arguments from below and then do `args << "NO_PKGCONFIG=y" << "CONTRIB_SDL=y" << "TILES=y" if build.with? "tiles"`